src/layout.ts

```diff
@@ -39,28 +39,32 @@ export function layoutPage(content: Paragraph[], box: Box, fonts: Font[]): Frame
   const { x, y, width, height } = box;
   const children = [];
   const pos = { x: 0, y: 0 };
+  let lastMargin = 0;
   let remainingHeight = height;
-  content.forEach((el) => {
-    const parPos = { x: pos.x, y: pos.y };
-    const parSize = { width, height: remainingHeight };
-    const frame = layoutParagraph(el, { ...parPos, ...parSize }, fonts);
+  content.forEach((paragraph) => {
+    const margin = parseEdges(paragraph.margin ?? 0);
+    const topMargin = Math.max(lastMargin, margin.top);
+    lastMargin = margin.bottom;
+    const nextPos = { x: pos.x + margin.left, y: pos.y + topMargin };
+    const maxSize = { width: width - margin.left - margin.right, height: remainingHeight };
```

Comment:
I wonder if the vertical margin of the current paragraph should be considered while computing the remaining height?

Reply:
Good point. I think you're right. The remaining height does not matter yet, but it will be needed to calculate page breaks. I've included the bottom margin as well since this space is reserved and not available for subsequent content.